src/lib.rs

```diff
@@ -9,6 +9,26 @@
 //!   structures.
 //!
 
+// Declares a 'const' member of a test module that's a new ExtensionId.
+// field_name should be an UPPERCASE i.e. name of a const.
+// This is here so that it is usable throughout the crate, but is not
+// visible from outside (as it would be if we used macro_export)
+// - as it won't *work* from outside because of the call to new_unchecked.
+#[allow(unused_macros)] // Because not used *in this file*
+macro_rules! test_const_ext_id {
```

Comment:
This is quite confusing at the call-sites since there is no indication whatsoever that we are declaring a variable.

Since this is an internal macro, I'd very much prefer to leave the declaration explicit, and require us to write checks on the side. We can even use a macro to make it a one-liner;
```rust
macro_rules! test_extension_id {
    ($field_name:ident) => {
        paste::paste! {
            #[test]
            fn [<check_ $field_name:lower _wellformed>]() {
                ExtensionId::new(&*$field_name).unwrap();
            }
        }
    };
}
```

And if we were to make this public, I'd be even more against hiding the `const` in the user-facing API

Reply:
Unfortunately `#[cfg(test)]` is only internal to the crate, it is never exported in the public API.

Your declaration inside the macro call is very much possible with macros by example (by exactly matching a const declaration), I think that's a good solution.

Reply:
Ok, so I now have a `pub fn new_unchecked` and am biting my tongue in protest :-( :-( :-(

However, the rest works well (and making `new_unchecked` pub was necessary for the doctest of the macro itself, which also validates the `$crate` etc. is correct).

For the record, I am totally happy to keep this private to the crate, i.e. just revert 82c9b3dbc42f9fe46a274dbb0feb07f00e93e353